core/src/modules/mappingHelper.ts

```diff
@@ -97,6 +115,12 @@ export namespace MappingHelper {
         );
       }
 
+      if (otherSourcePrimaryKey && property.sourceId === instance.id) {
+        throw new Error(
+          `'${instance.name}' cannot map '${remoteKey}' to own Property '${key}'. '${source.name}' must map to a Property from primary Source '${otherSourcePrimaryKey.source.name}'.`
+        );
+      }
```

Comment:
I'm still thinking about the wording. I think in the case this happen the primary source is ambiguous. For example if the Users and Purchases source point to their own properties, it's not clear which is the primary source. I'm open for feedback.

Maybe it should just say while {sourceB} is mapping to its own property {propertyB} ?

Reply:
What about something like

```ts
`${otherSourcePrimaryKey.soruce.name} is the primary source for this model - ${instance.name} requires a mapping back to ${otherSourcePrimaryKey.soruce.name}`
```

> so `users-table is the primary source from this model - purchases-table requires a mapping back to users-tabke`